numpy/core/src/umath/string_ufuncs.cpp

```diff
@@ -355,6 +401,76 @@ init_comparison(PyObject *umath)
 }
 
 
+static NPY_CASTING
+string_addition_resolve_descriptors(
+        PyArrayMethodObject *NPY_UNUSED(self),
+        PyArray_DTypeMeta *NPY_UNUSED(dtypes[3]),
+        PyArray_Descr *given_descrs[3],
+        PyArray_Descr *loop_descrs[3],
+        npy_intp *NPY_UNUSED(view_offset))
+{
+    Py_INCREF(given_descrs[0]);
+    loop_descrs[0] = NPY_DT_CALL_ensure_canonical(given_descrs[0]);
+    Py_INCREF(given_descrs[1]);
```

Comment:
Just realized that these are missing the error check.  Also the incref is clearly incorrect!